xtable-utilities/src/main/java/org/apache/xtable/utilities/RunSync.java

```diff
@@ -149,24 +150,45 @@ public static void main(String[] args) throws IOException {
           "Running sync for basePath {} for following table formats {}",
           table.getTableBasePath(),
           tableFormatList);
-      PerTableConfig config =
-          PerTableConfigImpl.builder()
-              .tableBasePath(table.getTableBasePath())
-              .tableName(table.getTableName())
+      Properties sourceProperties = new Properties();
+      if (table.getPartitionSpec() != null) {
+        sourceProperties.put(
```

Comment:
nit: This property should be needed required if the sourceFormat is Hudi

Reply:
it's not required if the table is non-partitioned. I have another set of changes to remove the need for this config altogether that I can open source in the future